lib/workers/repository/update/pr/index.ts

```diff
@@ -290,18 +290,46 @@ export async function ensurePr(
         }
       } else if (logJSON.error === 'MissingGithubToken') {
         upgrade.prBodyNotes ??= [];
-        upgrade.prBodyNotes = [
-          ...upgrade.prBodyNotes,
+        upgrade.prBodyNotes.push(
           [
             '> :exclamation: **Important**',
             '> ',
             '> Release Notes retrieval for this PR were skipped because no github.com credentials were available. ',
             '> If you are self-hosted, please see [this instruction](https://github.com/renovatebot/renovate/blob/master/docs/usage/examples/self-hosting.md#githubcom-token-for-release-notes).',
             '\n',
           ].join('\n'),
-        ];
+        );
       }
     }
+
+    const { packageName, depName, currentVersion, newVersion } = upgrade;
```

Comment:
can you move this to a separate function, so this function doesn't get too complex?